docs/commands.md

```diff
@@ -0,0 +1,153 @@
+
+Similar `kn` command: `kn service create NAME --image IMAGE [flags]`. This command allows a user to deploy a Knative Service by specifying an image, typically one hosted on a public container registry such as docker.io. The deployment options which the `kn` command affords the user are quite broad. The `kn` command in this case is quite effective for a power user. The `faas deploy` command has a similar end result, but is definitely easier for a user just getting started to be successful with.
+
+```console
+faas deploy [-n <namespace> -p <path>]
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas deploy [-n <namespace> -p <path>]
+```
+
+## `update`
+
+Updates the deployed Function project in the current directory. The user may specify the path on the command line with a flag. Reads the `.faas.yaml` configuration file to determine the image name. Derives the service name from the project name. The deployed Function is updated with a new container image that is pushed to a user repository, and the Knative `Service` is then updated.
+
+The namespace defaults to the value in `.faas.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specified on the command line, and if so this will overwrite the value in `.faas.yaml`. The user may specify a repository on the command line.
+
+Note that the behavior of `update` is different than that of `deploy` and `run`.  When `update` is run, a new container image is always built. However, for `deploy` and `run`, the user is required to run `faas build` first. The `update` command also differs from `deploy` in that it allows the user to specify a repository on the command line (but still not an image name). Consider normalizing all of this so that all of these commands behave similarly.
+
+Similar `kn` command: `kn service update NAME [flags]`. As with `deploy`, the `update` command provides a level of simplicity for a new user that restricts flexibility while improving the ease of use.
+
+```console
+faas update [-r <repository> -p <path>]
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas update [-r <repository> -p <path>]
+```
+
+## `describe`
+
+Prints the name, route and any event subscriptions for a deployed Function. The user may also specify the name of the function to describe, and the namespace in which to search for the function. If provided, this will be updated in the project's `.faas.yaml` configuration file. The user may also supply a path to the Function project.
+
+Similar `kn` command: `kn service describe NAME [flags]`. This flag provides a lot of nice information not available in `faas describe`, such as revisions, age, annotations and labels. Consider a way to combine the two outputs, or rename this command to make it distinct from `kn` - e.g. `faas info`.
```

Comment:
This is a good suggestion.  I would perhaps rename it to`status`, because that word implies info with the added property that it is transient state.  "the current status of the Function".  In particular I am thinking how we will likely want to add the current number of instances, and aggregate resource usage.  

The output of this `status` command (or `info`, which works too!) could perhaps be thought of as a reflection of the features the client provides. The current status of the "knobs" that `faas` allows to be turned.  So the `describe` of `kn` has more information, because it is showing the status/state of all the knobs that `kn` provides.  We provide less knobs, as we are trying to create a clean and simple abstraction.

However, there _are_ some bits of status which should bubble up from the underlying layer.   Information radiators, rather than knobs. As previously mentioned, number of instances and aggregate resource usage would be the top of my wish-list.

Reply:
I like the idea of `status` or `info`, great idea @lkingland !